bindings/pybind11/CMakeLists.txt

```diff
@@ -1,3 +1,26 @@
+set(libraryname idyntree-pybind11)
+
+add_library(${libraryname} INTERFACE)
+target_link_libraries(${libraryname} INTERFACE idyntree-core)
+
+# Specify include directories for both compilation and installation process.
+# The $<INSTALL_PREFIX> generator expression is useful to ensure to create
+# relocatable configuration files, see https://cmake.org/cmake/help/latest/manual/cmake-packages.7.html
+# creating-relocatable-packages
+target_include_directories(${libraryname} INTERFACE
+  "$<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/include>"
+  "$<INSTALL_INTERFACE:$<INSTALL_PREFIX>/${CMAKE_INSTALL_INCLUDEDIR}>")
+
+# Specify installation targets, typology and destination folders.
+install(TARGETS        ${libraryname}
+  EXPORT               ${PROJECT_NAME})
+
+install(FILES "${CMAKE_CURRENT_SOURCE_DIR}/include/iDynTree/pybind11/VectorCasters.h"
+  DESTINATION "${CMAKE_INSTALL_INCLUDEDIR}/iDynTree/pybind11")
+
+add_library(iDynTree::${libraryname} ALIAS ${libraryname})
+set_property(GLOBAL APPEND PROPERTY ${VARS_PREFIX}_TARGETS ${libraryname})
+
```

Comment:
tl;dr; Why these changes?

If I understood correctly, you want to create a separate library for the caster? Then
1) The name is very confusing. It does not explain what the library is for and it is a permutation of the other library name. 
2) `INTERFACE` library, no sources? I know this is ok, but I think adding the (single) header file and setting it as public header you can skip the `install FILES` command and make it more clear
3) Which should probably be 0): Why? Why do you need the header file outside the build directory? Do you have C++ libraries that depend on this?

Reply:
> INTERFACE library, no sources? I know this is ok, but I think adding the (single) header file and setting it as public header you can skip the install FILES command and make it more clear

In this specific case you can't as there destination path is `iDynTree/pybind11`, while the PUBLIC header stuff only support one folder as destination path.

Reply:
> The name is very confusing. It does not explain what the library is for and it is a permutation of the other library name.

We can change it

> INTERFACE library, no sources? I know this is ok, but I think adding the (single) header file and setting it as public header you can skip the install FILES command and make it more clear

I will need it to import it in a different project that uses idyntree types in the interface. Having an exported interface library simplifies importing it. It is exactly the same as `#include <pybind11/eigen.h>`

Reply:
> We can change it, what do you suggest?

Reply:
Can you also:
- move the install part down together with the install of the module?
- What is the purpose of the `ALIAS`?

Reply:
@GiulioRomualdi what do you think of `idyntree-pybind11-vector-casters` @GiulioRomualdi ? It seems a nice name to me.